From e66485f061bb8960a93e77ef2ad03ec0c5f466b4 Mon Sep 17 00:00:00 2001
From: Anson Huang <Anson.Huang@nxp.com>
Date: Wed, 23 Nov 2016 19:17:48 +0800
Subject: [PATCH 1289/5242] MLK-13524 ARM: imx: enable RBB for i.mx7ulp VLPS
 mode

commit  9091cf798852505b4e40031cf87a8b1fd576c540 from
https://source.codeaurora.org/external/imx/linux-imx.git

RBB needs to be enabled for VLPS mode to save
power, it can save ~0.4mA on VDD_DIG1.

Signed-off-by: Anson Huang <Anson.Huang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/mach-imx/pm-imx7ulp.c |    8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/arch/arm/mach-imx/pm-imx7ulp.c b/arch/arm/mach-imx/pm-imx7ulp.c
index fb8b0d8..bdf3a63 100644
--- a/arch/arm/mach-imx/pm-imx7ulp.c
+++ b/arch/arm/mach-imx/pm-imx7ulp.c
@@ -70,6 +70,8 @@
 #define BM_PMCTRL_RUNM		(3 << 8)
 #define BM_PMCTRL_STOPM		(7 << 0)
 
+#define BM_VLPS_RBBEN		(1 << 28)
+
 #define BM_CTRL_LDOEN		(1 << 31)
 #define BM_CTRL_LDOOKDIS	(1 << 30)
 
@@ -420,10 +422,16 @@ static int imx7ulp_pm_enter(suspend_state_t state)
 	switch (state) {
 	case PM_SUSPEND_STANDBY:
 		imx7ulp_set_lpm(VLPS);
+		writel_relaxed(
+			readl_relaxed(pmc1_base + PMC_VLPS) | BM_VLPS_RBBEN,
+			pmc1_base + PMC_VLPS);
 
 		/* Zzz ... */
 		cpu_suspend(0, imx7ulp_suspend_finish);
 
+		writel_relaxed(
+			readl_relaxed(pmc1_base + PMC_VLPS) & ~BM_VLPS_RBBEN,
+			pmc1_base + PMC_VLPS);
 		imx7ulp_set_lpm(RUN);
 		break;
 	case PM_SUSPEND_MEM:
-- 
1.7.9.5

